### ISSUE TITLE
nuxt seo kit doesnt listen to route rules with trailing slash enabled

### ISSUE THREAD

**Issue by jpengelbrecht** [harlan-zw/nuxt-seo-kit#140]
### Describe the bug

adding 
```
  site: {
    trailingSlash: true,
  },
```
breaks 
```
  routeRules: {
    "/maintenance/": { index: false },
    "/maintenance": { index: false },

  },
```

### Reproduction

https://github.com/jpengelbrecht/nuxt-seo-route-rules-issue/blob/main/nuxt.config.ts

- npm run generate
- go to output -> sitemap.xml
- see that maintenance was indexed

**Comment by jpengelbrecht** [comment]
@harlan-zw is there a way you can look into this one at some point?   think its quite a big one,  as the new SEO kit route rules get ignored when trailing slashes are enabled.

**Comment by harlan-zw** [comment]
Thanks for the ping, I'll try and look at this tomorrow.

**Comment by jpengelbrecht** [comment]
hero! thanks @harlan-zw

**Comment by harlan-zw** [comment]
I have fixed this in the sitemap module, a fix is available in [4.1.14](https://github.com/harlan-zw/nuxt-simple-sitemap/releases/tag/v4.1.14).

I'll be going through the other modules and making sure it's fixed for them as well.

Let me know if you run into any other issues though :)
